packages/amplify-category-auth/src/provider-utils/awscloudformation/handlers/get-add-auth-handler.ts

```diff
@@ -0,0 +1,30 @@
+import { ServiceQuestionsResult } from '../service-walkthrough-types';
+import { getAddAuthDefaultsApplier } from '../utils/add-auth-defaults-applier';
+import { getResourceSynthesizer } from '../utils/synthesize-resources';
+import { getPostAddAuthMetaUpdater } from '../utils/post-add-auth-meta-update';
+import { getPostAddAuthMessagePrinter } from '../utils/post-add-auth-message-printer';
+
+/**
+ * Factory function that returns a ServiceQuestionsResult consumer that handles all of the resource generation logic
+ * @param context The amplify context
+ */
+export const getAddAuthHandler = (context: any) => async (request: ServiceQuestionsResult) => {
+  const serviceMetadata = require('../../supported-services').supportedServices[request.serviceName];
+  const { cfnFilename, defaultValuesFilename, provider } = serviceMetadata;
+  let projectName = context.amplify.getProjectConfig().projectName.toLowerCase();
+  const disallowedChars = /[^A-Za-z0-9]+/g;
+  projectName = projectName.replace(disallowedChars, '');
+  return getAddAuthDefaultsApplier(
```

Comment:
`then` again.

Reply:
same as above, I think then makes sense here since we are pipelining these functions